models/cut_model.py

```diff
@@ -460,6 +460,34 @@ def data_dependent_initialize_semantic_mask(self, data):
             visual_names_out_mask_A = ["real_A_out_mask", "fake_B_out_mask"]
             self.visual_names += [visual_names_out_mask_A]
 
+    def inference(self):
```

Comment:
This is very similar to `forward_cut`, do we want to duplicate it ? Having key inference code in a single place is easier to maintain.